### ISSUE TITLE
Is it possible to add a heading to datagrid export?

### ISSUE THREAD

**Issue by mingvst** [mui/mui-x#11969]
### Order ID or Support key 💳

79227

### Search keywords

datagrid export pdf text

### Latest version

- [X] I have tested the latest version

### The problem in depth

We are having issues figuring out how to add text fields to an export of mui components, specifically adding a heading to the datagrid export, with the intention of supplying a PDF file to our customers. 

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

  System:
    OS: macOS 14.3
  Binaries:
    Node: 21.1.0 - /opt/homebrew/bin/node
    npm: 10.2.0 - /opt/homebrew/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: 121.0.6167.139
    Edge: Not Found
    Safari: 17.3
  npmPackages:
    @emotion/react: ^11.11.3 => 11.11.3 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.30 
    @mui/core-downloads-tracker:  5.15.3 
    @mui/icons-material: ^5.15.3 => 5.15.3 
    @mui/material: ^5.15.3 => 5.15.3 
    @mui/private-theming:  5.15.3 
    @mui/styled-engine:  5.15.3 
    @mui/system:  5.15.3 
    @mui/types:  7.2.12 
    @mui/utils:  5.15.3 
    @mui/x-charts: ^6.18.4 => 6.18.4 
    @mui/x-data-grid:  6.18.6 
    @mui/x-data-grid-premium: ^6.18.6 => 6.18.6 
    @mui/x-data-grid-pro:  6.18.6 
    @mui/x-date-pickers: ^6.18.6 => 6.19.0 
    @mui/x-date-pickers-pro: ^6.19.0 => 6.19.0 
    @mui/x-license-pro: ^6.10.2 => 6.10.2 
    @types/react: ^18.2.46 => 18.2.46 
    react: ^18 => 18.2.0 
    react-dom: ^18 => 18.2.0 
    typescript: ^5.3.3 => 5.3.3 

</details>

**Comment by michelengelen** [comment]
AFAIK It is currently not possible to add custom elements to the print export.
This seems like a nice feature to have, so I'll add the "waiting for upvotes" label.

**Comment by mingvst** [comment]
Thank you for the clarification, heres to hoping it will eventually get implemented :)

**Comment by flaviendelangle** [comment]
I think it should be possible to manually add those headings using the `exceljsPostProcess` prop
You have a very basic doc example [here](https://mui.com/x/react-data-grid/export/#customization)

**Comment by michelengelen** [comment]
> I think it should be possible to manually add those headings using our `exceljsPostProcess` You have a very basic doc example [here](https://mui.com/x/react-data-grid/export/#customization)

But this is for excel only, right?
@mingvst was asking about a PDF file and thats vie print ony, right? 🤔

**Comment by flaviendelangle** [comment]
Oh sorry I read too fast...
Maybe we could provide a similar API for PDF and/or print

**Comment by michelengelen** [comment]
> Oh sorry I read too fast... Maybe we could provide a similar API for PDF and/or print

Yep, I thought the same. That's basically why I added the "waiting for upvotes" label.
